errors.js

```diff
@@ -1,18 +1,52 @@
 'use strict'
 
-const createError = require('errno').create
-const LevelUPError = createError('LevelUPError')
-const NotFoundError = createError('NotFoundError', LevelUPError)
+function createError (type, Proto) {
+  const Err = function (message, cause) {
+    if (typeof message === 'object' && message !== null) {
+      // Can be passed just a cause
+      cause = cause || message
+      message = message.message || message.name
+    }
```

Comment:
This seems unnecessarily magic, can we make it clearer, like this?

```suggestion
    if (typeof message === 'object' && message !== null) {
      cause = message
      message = message.message
    }
```

Since we control the error creation, I think we can simplify the API and only support `(message, causeeErr)` and `(causeErr)`. Also, why would you want to read `message` from `.name` ?

Reply:
Agree on both points, but I'd like to postpone this cleanup so that we can release this PR as semver-patch.

Reply:
https://github.com/Level/errors/issues/38